lib/ut-coverage/test_v7_dataservice_api.c

```diff
@@ -111,324 +106,361 @@ void Test_bplib_dataservice_detach(void)
-    UtAssert_UINT32_EQ(bplib_recv(desc, payload, &size, timeout), 0);  
-            
-    if (sock.parent_rtbl)
-    {
-        free(sock.parent_rtbl);
-    }        
-    if (desc)
-    {
-        free(desc);
-    }
+    UtAssert_UINT32_EQ(bplib_recv(&desc, payload, &size, timeout), 0);  
+        
+    UT_SetHandlerFunction(UT_KEY( bplib_mpool_bblock_cbor_export), UT_lib_sizet_1_Handler, NULL);
+    UtAssert_UINT32_NEQ(bplib_recv(&desc, payload, &size, timeout), 0);  
+    
+    ccb_pay.encoded_content_offset = 0;
+    UtAssert_UINT32_NEQ(bplib_recv(&desc, payload, &size, timeout), 0);  
+
+    UT_SetHandlerFunction(UT_KEY(bplib_mpool_generic_data_cast), UT_AltHandler_PointerReturn, NULL);
+    UT_SetHandlerFunction(UT_KEY(bplib_mpool_flow_cast), UT_AltHandler_PointerReturn, NULL);     
+    UT_SetHandlerFunction(UT_KEY(bplib_mpool_flow_try_pull), UT_AltHandler_PointerReturn, NULL);   
+    UT_SetHandlerFunction(UT_KEY(bplib_mpool_ref_from_block), UT_AltHandler_PointerReturn, NULL);       
+    UT_SetHandlerFunction(UT_KEY(bplib_mpool_bblock_primary_cast), UT_AltHandler_PointerReturn, NULL);   
 }
 
 /*int bplib_serviceflow_forward_ingress(void *arg, bplib_mpool_block_t *subq_src)*/
 void Test_bplib_serviceflow_forward_ingress(void)
 {
-    bplib_routetbl_t *tbl = calloc(1, 1<<20);
+    bplib_routetbl_t tbl;
+    memset(&tbl, 0, sizeof(bplib_routetbl_t));
     bplib_mpool_block_t subq_src;
     memset(&subq_src, 0, sizeof(bplib_mpool_block_t));
     bplib_mpool_block_t sblk;
     memset(&sblk, 0, sizeof(bplib_mpool_block_t));      
+    pthread_t thread1;
     
-    UT_SetHandlerFunction(UT_KEY(bplib_mpool_flow_try_pull), UT_AltHandler_PointerReturn, NULL);
     UT_SetHandlerFunction(UT_KEY(bplib_mpool_get_block_from_link), UT_lib_sizet_Handler, NULL);      
-    UT_SetHandlerFunction(UT_KEY(bplib_mpool_generic_data_cast), UT_AltHandler_PointerReturn, NULL);
-    UtAssert_UINT32_EQ(bplib_serviceflow_forward_ingress(tbl, &subq_src), 0); 
+    UtAssert_UINT32_EQ(bplib_serviceflow_forward_ingress(&tbl, &subq_src), 0); 
     
     UT_SetHandlerFunction(UT_KEY(bplib_mpool_generic_data_cast), UT_AltHandler_PointerReturn, &sblk);    
     UT_SetHandlerFunction(UT_KEY(bplib_mpool_flow_cast), UT_AltHandler_PointerReturn, NULL);
-    UtAssert_UINT32_NEQ(bplib_serviceflow_forward_ingress(tbl, &subq_src), 0); 
-    
+    UtAssert_UINT32_NEQ(bplib_serviceflow_forward_ingress(&tbl, &subq_src), 0);     
 
     UT_SetHandlerFunction(UT_KEY(bplib_mpool_flow_cast), UT_AltHandler_PointerReturn, &sblk);
-    UtAssert_UINT32_EQ(bplib_serviceflow_forward_ingress(tbl, &subq_src), 0); 
+    UtAssert_UINT32_EQ(bplib_serviceflow_forward_ingress(&tbl, &subq_src), 0); 
     
-    /* Don't know how to get out of the while loop*/
-    //UT_SetHandlerFunction(UT_KEY(bplib_mpool_flow_try_push), V7_Dataservice_Bool_Handler, NULL); 
+    pthread_create( &thread1, NULL, SetPullReturnValue, NULL);
```

Comment:
Can't use `pthread_create()` from a unit test, because these need to run on other (non-posix) platforms too.  This is also basically a race condition - a heavily weighted one, but a race nonetheless.

If the objective is to make it return a non-NULL until some point where it then returns NULL, can the call counter be used?  That is, if (call count < n) return ptr; else return NULL?